src/psyclone/domain/gocean/transformations/gocean_opencl_trans.py

```diff
@@ -932,9 +932,8 @@ def _insert_ocl_arg_setter_routine(node, kernel):
             # will generate it.
             pass
 
-        # Create the new Routine and RoutineSymbol
-        node.symbol_table.add(RoutineSymbol(sub_name), tag=sub_name)
-        argsetter = Routine(sub_name)
+        # Create the new Routine
+        argsetter = Routine(sub_name, symbol_tag=sub_name)
```

Comment:
I fixed this now by just doing:
`sub_name = node.symbol_table.next_available_name(sub_name)`

Reply:
Yes - but not if that symbol is already in the symbol table (because there was no current method to distinguish that behaviour and attempting to place a Routine somewhere where the symbol for it already exists).